x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/util/MathUtils.java

```diff
@@ -14,6 +14,11 @@ public class MathUtils {
     public static int abs(int number) {
 
         if (number == Integer.MIN_VALUE) {
+            // TODO: can this function be removed?
```

Comment:
Given the code that uses this method I would say it's not possible, but it's not so easy to evaluate this because there are some additions that happen. I would keep this code.

Nevertheless, I've went ahead and did some tests and there are some issues with code that sits before this method, more specifically `head` and `tail` pipes. I've created an issue [here](https://github.com/elastic/elasticsearch/issues/101724).

Reply:
> Given the code that uses this method I would say it's not possible, but it's not so easy to evaluate this because there are some additions that happen.

When I looked over this, I thought that if we switched those additions to exact ones, this check would no longer be needed (and thus nor the method). In any case, for the purpose of this PR, keeping the exception as is in place makes sense.

Thanks for checking!